doc/introduction/compiling_circuits.rst

```diff
@@ -0,0 +1,252 @@
+
+.. code-block:: python
+
+    def circuit(weights):
+        qml.BasicEntanglerLayers(weights, wires=[0, 1, 2])
+        return qml.expval(qml.PauliZ(0))
+
+    original_dev = qml.device("default.qubit", wires=3)
+    original_qnode = qml.QNode(circuit, original_dev)
+
+>>> weights = np.array([[0.4, 0.5, 0.6]])
+>>> print(qml.draw(original_qnode, expansion_strategy="device")(weights))
+0: ──RX(0.4)──╭C──────╭X──┤ ⟨Z⟩
+1: ──RX(0.5)──╰X──╭C──│───┤
+2: ──RX(0.6)──────╰X──╰C──┤
+
+Now, let's swap out PennyLane's default decomposition of the ``CNOT`` gate into ``CZ``
+and ``Hadamard``.
+We define the custom decompositions like so, and pass them to a device:
+
+.. code-block:: python
+
+    def custom_cnot(wires):
+        return [
+            qml.Hadamard(wires=wires[1]),
+            qml.CZ(wires=[wires[0], wires[1]]),
+            qml.Hadamard(wires=wires[1])
+        ]
+
+    custom_decomps = {qml.CNOT: custom_cnot}
+
+    decomp_dev = qml.device("default.qubit", wires=3, custom_decomps=custom_decomps)
+    decomp_qnode = qml.QNode(circuit, decomp_dev)
+
+Now when we draw or run a QNode on this device, the gates will be expanded
+according to our specifications:
+
+>>> print(qml.draw(decomp_qnode, expansion_strategy="device")(weights))
+0: ──RX(0.40)────╭C──H───────╭Z──H─┤  <Z>
+1: ──RX(0.50)──H─╰Z──H─╭C────│─────┤
+2: ──RX(0.60)──H───────╰Z──H─╰C────┤
+
+.. note::
+    If the custom decomposition is only supposed to be used in a specific code context,
+    a separate context manager :func:`~.pennylane.transforms.set_decomposition` can be used.
+
+Circuit cutting
+---------------
+
+Circuit cutting allows you to replace a circuit with ``N`` wires by a set of circuits with less than
+``N`` wires (see also `Peng et. al <https://arxiv.org/abs/1904.00102>`_). Of course this comes with a cost: The smaller circuits
+require a greater number of device executions to be evaluated.
+
+In PennyLane, circuit cutting can be
+activated by positioning :class:`~.pennylane.WireCut` operators at the desired cut locations, and
+by decorating the QNode with the :func:`~.pennylane.cut_circuit` transform.
+
+The example below shows how a three-wire circuit can be run on a two-wire device:
+
+.. code-block:: python
+
+    dev = qml.device("default.qubit", wires=2)
+
+    @qml.cut_circuit
+    @qml.qnode(dev)
+    def circuit(x):
+        qml.RX(x, wires=0)
+        qml.RY(0.9, wires=1)
+        qml.RX(0.3, wires=2)
+
+        qml.CZ(wires=[0, 1])
+        qml.RY(-0.4, wires=0)
+
+        qml.WireCut(wires=1)
+
+        qml.CZ(wires=[1, 2])
+
+        return qml.expval(qml.grouping.string_to_pauli_word("ZZZ"))
+
+Instead of executing the circuit directly, it will be partitioned into
+smaller fragments according to the :class:`~.pennylane.WireCut` locations,
+and each fragment executed multiple times. PennyLane automatically combines the results
+of the fragment executions to recover the expected output of the original uncut circuit.
```

Comment:
```suggestion
Instead of being executed directly, the circuit will be partitioned into
smaller fragments according to the :class:`~.pennylane.WireCut` locations,
and each fragment will be executed multiple times. PennyLane automatically combines the results
of the fragment executions to recover the expected output of the original uncut circuit.
```